tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchTextBuilderFacts.cs

```diff
@@ -0,0 +1,170 @@
+                    // See: https://github.com/NuGet/NuGetGallery/issues/6922
+                    { "id:Hello packageId:World", "packageId:(Hello world)" },
+                    { "owner:A owners:B", "owners:(a b)" },
+
+                    // The version field is normalized, if possible, and lowercased
+                    { "version:1.0.0.0", "normalizedVersion:1.0.0" },
+                    { "version:1.0.0.0-TEST", "normalizedVersion:1.0.0\\-test" },
+                    { "version:Thisisnotavalidversion", "normalizedVersion:thisisnotavalidversion" },
+
+                    // The tags field is split by delimiters
+                    { "tag:a,b;c|d", "tags:(a b c d)" },
+                    { "tags:a,b;c|d", "tags:(a b c d)" },
+
+                    // Field-scoped terms on the same field are OR'd together
+                    // TODO: id and packageId query fields should map to different fields with different analyzers
+                    // See: https://github.com/NuGet/NuGetGallery/issues/6920
+                    // See: https://github.com/NuGet/NuGetGallery/issues/6922
+                    { "id:foo id:bar", "packageId:(foo bar)" },
+                    { "packageId:foo packageId:bar", "packageId:(foo bar)" },
+                    { "title:hello title:world", "title:(hello world)" },
+                    { "description:I description:am", "description:(I am)" },
+                    { "tag:a tag:sentient tags:being", "tags:(a sentient being)" },
+                    { "author:a author:b authors:c", "authors:(a b c)" },
+                    { "summary:d summary:e", "summary:(d e)" },
+                    { "owner:billy owners:the owner:goat", "owners:(billy the goat)" },
+                    { @"tag:a,b;c tags:d tags:""e f""", "tags:(a b c d e f)" },
+                    
+                    // Field-scoped terms on different fields are AND'd together
+                    { "title:foo description:bar title:baz", "title:(foo baz) AND description:bar" },
+                    { "title:foo unknown:bar", "title:foo" },
+
+                    // If there are non-field-scoped terms and no field-scoped terms, at least of one the non-field-scoped terms is required.
+                    { "foo bar", "+(foo bar)" },
+                    { "id packageId version title description tag author summary owner owners", "+(id packageId version title description tag author summary owner owners)" },
+                    
+                    // Quotes allow adjacent terms to be searched
+                    { @"""foo bar""", @"+(""foo bar"")" },
+                    { @"""foo bar"" baz", @"+(""foo bar"" baz)" },
+                    { @"id:""foo bar""", @"packageId:""foo bar""" },
+                    { @"id:""a b"" c id:d packageId:e f", @"+packageId:(""a b"" d e) c f" },
+
+                    // Duplicate search terms on the same query field are folded
+                    { "a a", "+(a)" },
```

Comment:
My knowledge about non-scope Lucene queries is not strong enough. Do you really need the `+` when there is only one term?

Reply:
Yes, the spec explicitly called this behavior out so I made it explicit in the generated queries as well. I agree it's a little ugly, so I removed the + when from non-field-scoped terms